### ISSUE TITLE
fail typechecking if a parameterized type doesn't declare `__class_getitem__`

### ISSUE THREAD

**Issue by 3n-k1** [python/mypy#9712]
**feature**

a parameterized (generic) class/type should fail to typecheck if it doesn't have the `__class_getitem__` method

**pitch**

the syntax for parameterized types is `SomeClass[int]`. however, if `SomeClass` doesn't define the `__class_getitem__` method, this will typecheck but not actually run properly. we ran into this unexpectedly over at typeddjango/django-stubs/issues/507

it would be *extremely* helpful to at least have a command line option to warn/error on this, just so that nobody runs into it unexpectedly

**Comment by 3n-k1** [comment]
after some further investigation it seems this issue only pops up when using generics in stub files, but not in source. here's a minimal example i cooked up:

```
.
├── lib.py
├── lib.pyi
└── main.py
```

```py
# lib.py
class ModelAdmin: pass

# lib.pyi
from typing import Generic, TypeVar

T = TypeVar("T")

class ModelAdmin(Generic[T]): ...

# main.py
from lib import ModelAdmin

class FooAdmin(ModelAdmin[type]): pass
```

it typechecks:
```
$ mypy --strict main.py 
Success: no issues found in 1 source file
```

but does not run:
```
$ python main.py 
Traceback (most recent call last):
  File "main.py", line 3, in <module>
    class FooAdmin(ModelAdmin[type]): pass
TypeError: 'type' object is not subscriptable
```

so it seems this problem is really only a show-stopper for pep 561 stub-only packages that are not maintained by the library maintainers. in django-stubs we're working on an extension package to monkeypatch the `__class_getitem__` method, but it would be extremely helpful to be able to have ci fail if a generic somewhere has an issue
